core-ui/src/components/Lambdas/LambdaDetails/Tabs/Code/LambdaVariables/LambdaVariables.js

```diff
@@ -119,19 +140,126 @@ function VariableValue({ variable }) {
   return value;
 }
 
+function SecretVariableValue({ variable }) {
+  const [show, setShow] = useState(false);
+
+  const { data, loading, error } = useQuery(GET_SECRET, {
+    variables: { namespace: variable.namespace, name: variable.resourceName },
+  });
+
+  if (loading) return null;
+  if (error) return `Error: ${error}`;
+
+  if (data.secret === null) {
+    return <span style={{ color: 'red' }}>{'The Secret does not exist!'}</span>;
+  }
+
+  if (data.secret.data[variable.resourceKey] === undefined) {
+    return (
+      <span style={{ color: 'red' }}>
+        {'There is no such key in the Secret!'}
+      </span>
+    );
+  }
+
+  const value = <span>{data.secret.data[variable.resourceKey] || '-'}</span>;
+
+  const blurVariable = (
+    <div className={!show ? 'blur-variable' : ''} onClick={_ => setShow(!show)}>
```

Comment:
```suggestion
    <div className={!show ? 'blur-variable' : ''} onClick={_ => setShow(state => !state)}>
```
That's an alternate way to handle flipping the state, by supplying a function that gets current state and returns new one. It's a good practice and prevent subtle bugs.